weather_mv/loader_pipeline/ee_util.py

```diff
@@ -0,0 +1,457 @@
+        """Writes tiff file, upload it to bucket and ingest into earth engine"""
+
+        # Initializing earth engine
+        if not self.dry_run:
+            ee_initialize(use_personal_account=self.use_personal_account)
+
+        file_name = f'{grib_data.name}.tiff'
+
+        with tempfile.TemporaryDirectory() as temp_dir:
+            local_tiff_path = os.path.join(temp_dir, file_name)
+            with rasterio.open(local_tiff_path, 'w',
+                               driver='GTiff',
+                               dtype=grib_data.dtype,
+                               width=grib_data.data[0].data.shape[1],
+                               height=grib_data.data[0].data.shape[0],
+                               count=len(grib_data.data),
+                               crs=grib_data.projection,
+                               transform=grib_data.transform,
+                               tiled=True) as f:
+                for i, da in enumerate(grib_data.data):
+                    f.write(da, i+1)
+
+            # Copy local tiff to gcs tiff location
+            storage_client = storage.Client()
+            parsed_tiff_location = urlparse(self.tiff_location)
+            bucket_name = parsed_tiff_location.netloc
+            bucket = storage_client.get_bucket(bucket_name)
+
+            tiff_internal_path = str(parsed_tiff_location.path)
+            tiff_internal_path = tiff_internal_path[1:] if tiff_internal_path.startswith('/') else tiff_internal_path
+            tiff_bucket_path = os.path.join(tiff_internal_path, file_name)
+
+            logger.debug(f'Copying {local_tiff_path} to {tiff_bucket_path}')
+            blob = bucket.blob(tiff_bucket_path)
+            blob.upload_from_filename(local_tiff_path)
+
+            if self.dry_run:
+                # In case of dry run, we'll not push tiff into earth engine.
+                yield tiff_bucket_path
+                return
+
+            # Push into earth engine
+            out_path = os.path.join(self.tiff_location, file_name)
+            asset_id = os.path.join(self.ee_asset, grib_data.name)
+            variable_list = [da.name for da in grib_data.data]
+            start_time = grib_data.start_time
+            end_time = grib_data.end_time
+            properties = grib_data.metadata
+
+            task_id = self.manifest_output(out_path, asset_id, variable_list, start_time, end_time, properties)
+            yield task_id
+
+
+def _cleanup(storage_client: storage.Client,
+             canay_bucket_name: str,
+             sig: t.Optional[t.Any] = None,
+             frame: t.Optional[t.Any] = None) -> None:
+    """Cleaning up the bucket."""
+    try:
+        storage_client.get_bucket(canay_bucket_name).delete(force=True)
+    except NotFound:
+        pass
+    if sig:
+        traceback.print_stack(frame)
+        sys.exit(0)
+
+
+def validate_region(temp_location: t.Optional[str] = None, region: t.Optional[str] = None) -> None:
```

Comment:
Is this the same as the BQ validation? If so, how about we move it to a shared place and re-use the method. 

If it's different, how can we abstract out the parts that change and re use the logic as much as possible?

Reply:
Right. Moved the validate_region to util.